cmd/buildctl/build/util.go

```diff
@@ -26,27 +25,26 @@ func loadGithubEnv(cache client.CacheOptionsEntry) (client.CacheOptionsEntry, er
 	}
 
 	if _, ok := cache.Attrs["url_v2"]; !ok && version == "2" {
-		url, ok := os.LookupEnv("ACTIONS_RESULTS_URL")
-		if !ok {
-			return cache, errors.New("cache with type gha requires url parameter or $ACTIONS_RESULTS_URL")
+		// https://github.com/actions/toolkit/blob/2b08dc18f261b9fdd978b70279b85cbef81af8bc/packages/cache/src/internal/config.ts#L34-L35
+		if v, ok := os.LookupEnv("ACTIONS_RESULTS_URL"); ok {
+			cache.Attrs["url_v2"] = v
```

Comment:
why skip the error if `ACTIONS_CACHE_SERVICE_V2` is set but `ACTIONS_RESULTS_URL` is not?

Reply:
This relates to https://github.com/moby/buildkit/pull/5754#discussion_r1959756775 so error would be handled from daemon side anyway.